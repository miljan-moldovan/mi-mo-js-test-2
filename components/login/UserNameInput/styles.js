import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  inputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingLeft: 20,
    paddingRight: 20,
    marginBottom: 3,
    justifyContent: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.15)',
    maxWidth: 311,
    borderRadius: 4,
  },
  inputInvalid: {
    borderStyle: 'solid',
    borderLeftWidth: 1,
    borderRightWidth: 1,
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: 'red',
  },
  inputIcon: {
    width: 16,
    height: 16,
    fontSize: 16,
    alignSelf: 'center',
    resizeMode: 'contain',
    color: 'rgba(255, 255, 255, 0.25)',
  },
  input: {
    color: 'white',
    fontSize: 12,
    fontWeight: '500',
    paddingTop: 0,
    marginTop: 0,
    marginLeft: 22,
    paddingLeft: 0,
  },
  inputFontFamily: {
    fontFamily: 'Roboto-Regular',
  },
  inputPlaceholderFontFamily: {
    fontFamily: 'Roboto-Italic',
  },
});

export default styles;
