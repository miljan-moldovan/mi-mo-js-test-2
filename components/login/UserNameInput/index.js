import React from 'react';
import FontAwesome, { Icons } from 'react-native-fontawesome';
import { Input } from 'native-base';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

function UserNameInput(props) {
  const {
    loggedIn,
    userNameError,
    username,
    handleUsernameChange,
  } = props;

  return (
    <View style={[styles.inputContainer, userNameError ? styles.inputInvalid : null]}>
      <FontAwesome style={styles.inputIcon}>{Icons.user}</FontAwesome>
      <Input
        style={[styles.input, username ?
          styles.inputFontFamily : styles.inputPlaceholderFontFamily]}
        underline={false}
        disabled={loggedIn}
        autoCorrect={false}
        blurOnSubmit
        autoCapitalize="none"
        onChangeText={handleUsernameChange}
        placeholder="Username"
        placeholderTextColor="rgb(76.15, 135.15, 216.75)"
      />
    </View>
  );
}

UserNameInput.propTypes = {
  handleUsernameChange: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired,
  userNameError: PropTypes.bool.isRequired,
};

export default UserNameInput;
