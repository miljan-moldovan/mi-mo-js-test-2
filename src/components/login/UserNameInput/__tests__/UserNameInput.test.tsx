import * as React from 'react';
import { shallow } from 'enzyme';

import UserNameInput from '../';

describe('UserNameInput', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <UserNameInput
        loggedIn
        handleUsernameChange={jest.fn()}
        username="test"
        userNameError
      />,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
