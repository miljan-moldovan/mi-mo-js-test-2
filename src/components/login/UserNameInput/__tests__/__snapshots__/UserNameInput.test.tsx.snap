// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNameInput renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.15)",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 3,
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 311,
        "paddingLeft": 20,
        "paddingRight": 20,
      },
      Object {
        "borderBottomWidth": 1,
        "borderColor": "red",
        "borderLeftWidth": 1,
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopWidth": 1,
      },
    ]
  }
>
  <Icon
    style={
      Object {
        "alignSelf": "center",
        "color": "rgba(255, 255, 255, 0.25)",
        "fontSize": 16,
        "height": 16,
        "resizeMode": "contain",
        "width": 16,
      }
    }
  >
    ï€‡
  </Icon>
  <Styled(Input)
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    disabled={true}
    onChangeText={[MockFunction]}
    placeholder="Username"
    placeholderTextColor="rgb(76.15, 135.15, 216.75)"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "500",
          "marginLeft": 22,
          "marginTop": 0,
          "paddingLeft": 0,
          "paddingTop": 0,
        },
        Object {
          "fontFamily": "Roboto-Regular",
        },
      ]
    }
    underline={false}
    value="test"
  />
</View>
`;
